tmt/export/__init__.py

```diff
@@ -430,131 +445,78 @@ def check_md_file_respects_spec(md_path: Path) -> list[str]:
-    count_html_headings(sections_headings['Cleanup'][0])
-
-    warn_outside_test_section = (
-        'Heading "{}" from the section "{}" is used \noutside of Test sections.'
-    )
-    warn_headings_not_in_pairs = (
-        'The number of headings from the section "Step" - {}\ndoesn\'t equal to the '
-        'number of headings from the section \n"Expect" - {} in the test section "{}"'
-    )
-    warn_required_section_is_absent = '"{}" section doesn\'t exist in the Markdown file'
-    warn_unexpected_headings = 'Headings "{}" aren\'t expected in the section "{}"'
-
-    def required_section_exists(
-        section: list[str], section_name: str, prefix: Union[str, tuple[str, ...]]
-    ) -> int:
-        res = list(filter(lambda t: t.startswith(prefix), section))
-        if not res:
-            warnings_list.append(warn_required_section_is_absent.format(section_name))
-            return 0
-        return len(res)
-
-    # Required sections don't exist
-    if not required_section_exists(html_headings_from_file, 'Test', '<h1>Test'):
-        return warnings_list
-
-    # Remove Optional heading #Cleanup if it's in the end of document
-    if html_headings_from_file[-1] == '<h1>Cleanup</h1>':
-        html_headings_from_file.pop()
-        # Add # Test heading to close the file
-        html_headings_from_file.append(sections_headings['Test'][0])
-
-    index = 0
-    while html_headings_from_file:
-        # # Step cannot be used outside of test sections.
-        if (
-            html_headings_from_file[index] == sections_headings['Step'][0]
-            or html_headings_from_file[index] == sections_headings['Step'][1]
-        ):
-            warnings_list.append(
-                warn_outside_test_section.format(html_headings_from_file[index], 'Step')
+    warnings = []
+    test_sections: list[TestSection] = []
+    current_test: Optional[TestSection] = None
+
+    for level, heading in headings:
+        section_type = get_heading_section(heading)
+
+        # invalid section
+        if not section_type:
+            warnings.append(f'Unknown html heading "{heading}" is used')
+            continue
+
+        # Start new test section on h1 heading
+        if level == 1:
+            if current_test:
+                test_sections.append(current_test)
+
+            current_test = TestSection(name=heading) if section_type == "Test" else None
+            continue
+
+        # Inside an open test section
+        if current_test:
+            if section_type == "Step":
+                current_test.steps.append(heading)
+            elif section_type == "Expect":
+                current_test.expects.append(heading)
+            else:
+                warnings.append(
+                    f'Heading "{heading}" isn\'t expected in the section "{current_test.name}"'
+                )
+        # Outside test section — detect orphan Step/Expect
+        elif section_type in ("Step", "Expect"):
```

Comment:
```suggestion
        elif section_type in {"Step", "Expect"}:
```
Eventually ruff will catch these, but it is not enabled yet.